text/0000-project-error-handling.md

```diff
@@ -0,0 +1,121 @@
+- Convert to a reporting type when the error is no longer expected to be handled beyond reporting e.g. `anyhow::Error` or `eyre::Report` or when trait object + downcast error handling is preferable.
+- Recommend `Box`ing concrete error types when stack size is an issue rather than `Box`ing and converting to `dyn Error`s.
+- What is the consensus on handling `dyn Error`s? Should it be encouraged or discouraged? Should we look into making `Box<dyn Error...>` impl Error?
+
+
+### Communicate current best practices
+
+- Document the concensus.
+- Communicate plan for future changes to error handling, and the libraries that future changes are being based off of.
+- Produce learning resources related to current best practices.
+    - New chapters in the book?
+
+### Evaluate options for error reporting type aka better `Box<dyn Error>`
+
+- Survey the current libraries in the ecosystem:
+    - `anyhow`
+    - `eyre`
+- Evaluate value of features including:
+    - Single word width on stack
+    - Error wrapping with display types and with special downcast support.
+    - report hook + configurable `dyn ReportHandler` type for custom report formats and content, similar to panic handler but for errors.
+    - libcore compatibility.
+
+### Consolidate ecosystem by merging best practice crates into std
+
+- Provide a derive macro for `Error` in std.
+- Stabilize the `Backtrace` type but possibly not `fn backtrace` on the Error trait.
+    - Provide necessary API on `Backtrace` to support crates like `color-backtrace`.
+- Move error to core .
+    - Depends on generic member access.
+    - Requires resolving downcast dependency on `Box` and blocking the stabilization of `fn backtrace`.
+- Potentially stabilize an error Reporting type based on `anyhow` and `eyre` now that they're close to having identical feature sets.
+
+### Add missing features
+
+- Generic member access on the `Error` trait.
+- `Error` return traces:
+    - Depends on specialization and generic member access.
+- Fix rough corners around reporting errors and `Termination`.
+
+## Non Goals
+
+- This group should not be involved in design discussions for the `Try` trait, `try` blocks, or `try` fns.
+
+## Membership Requirements
+
+- Group membership is open, any interested party can participate in discussions, repeat contributors will be added appropriate teams.
+
+## Additional Questions
+
+### What support do you need, and separately want, from the Rust organization?
+
+I'm not sure, my main concern is getting prompt feedback on RFCs.
+
+### Why should this be a project group over a community effort?
+
+Error handling has historically been handled via community effort which has resulted a great deal of competing and outdated solutions and resources. An officially supported project group has the best chance of writing and maintaining up-to-date resources that would reach a wide audience.
```

Comment:
I don't know that I fully agree with this declaration. 

- "a great deal of competing [...] solutions" is presented as a negative, but I don't see any details in this proposal to support the claim.
- "outdated [...] resources" — I'm not the target of these resources so I'm not familiar with any outdated resources. Could you provide some links?
- "An officially supported project group has the best chance of writing and maintaining up-to-date resources" — What prevents creating these resources, keeping them updated for "long enough", and then having them be adopted?

Since you said "solutions and resources", I'm interpreting it as solution == "a crate" and resource == "documentation". If that's not what you mean, you may wish to clarify.

Reply:
You raise good points, I'll try to re-evaluate the justification here. By outdated resources I was recalling https://blog.burntsushi.net/rust-error-handling/ but it's actually been updated this year, you're right that this seems to mostly be FUD.